consensus/src/state_computer.rs

```diff
@@ -155,18 +167,23 @@ impl StateComputer for ExecutionProxy {
         let skip_clean = blocks.is_empty();
         let mut latest_epoch: u64 = 0;
         let mut latest_round: u64 = 0;
+        let mut payloads = Vec::new();
 
         for block in blocks {
             block_ids.push(block.id());
-            txns.extend(block.transactions_to_commit(&self.validators.lock()));
-            reconfig_events.extend(block.reconfig_event());
 
-            if block.epoch() > latest_epoch {
-                latest_epoch = block.epoch();
-            }
-            if block.round() > latest_round {
-                latest_round = block.round();
+            if let Some(payload) = block.block().payload() {
+                payloads.push(payload.clone());
             }
+
+            debug!("QSE: getting data in commit, round {}", block.round());
```

Comment:
this is written always, not only when we are missing data, and need to fetch it in commit, as message suggests?

Reply:
At this point, we do not know if we are missing data. We ask PayloadManager to extract the data from the payload. If QuorumStore is enabled, we pre-fetch the data once we add the block to the tree. The pre-fetch code is not part of the PR, this is implemented in BatchReader and BatchRequester.

Reply:
we probably don't want to print debug messages like this, it's too frequent and also duplicated (I think we log committed somewhere)